### PR TITLE
Label image during build #681

### LINKED ISSUE #681: Label image during build
Hey,

just noticed that there's no way of adding labels to images that are build built with `buildImageCmd` (or am i missing something?) .

From https://docs.docker.com/engine/reference/api/docker_remote_api_v1.24/#/build-image-from-a-dockerfile we can pass `labels` so that at the end the labels are added to it. 

Maybe that could be a `.withLabels` or something like that. Wdyt?

Thx!

### PR DESCRIPTION
**Pull request by cberes**
Fixes #681

This change is [Reviewable](https://reviewable.io/reviews/docker-java/docker-java/684)

### CODE REVIEW

File: src/main/java/com/github/dockerjava/core/command/BuildImageCmdImpl.java [hunk 1]
```diff
@@ -303,6 +310,18 @@ public BuildImageCmd withShmsize(Long shmsize) {
         return this;
     }
 
+    /**
+     * @see #labels
+     */
+    @Override
+    public BuildImageCmd withLabel(String key, String value) {
+        if (this.labels == null) {
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/core/command/BuildImageCmdImpl.java]: this unneeded

**Comment by cberes** [comment]
Review status: 0 of 7 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[src/main/java/com/github/dockerjava/core/command/BuildImageCmdImpl.java, line 318 \[r1\]](https://reviewable.io:443/reviews/docker-java/docker-java/684#-KQEWHl3ssfwfNxrm7ha-r1-318:-KQGh5SgEI3IoJghpXsG:b-4pgkjn) ([raw file](https://github.com/docker-java/docker-java/blob/761f41655f9e01ed25552309b7dd3a23472a56c4/src/main/java/com/github/dockerjava/core/command/BuildImageCmdImpl.java#L318)):*
<details><summary><i>Previously, KostyaSha (Kanstantsin Shautsou) wrote…</i></summary>
> `this` unneeded
</details>
you  prefer to create the map when the object is created? for example:
private Map<String, String> labels = new HashMap<>();

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/docker-java/docker-java/684)*

**Comment by KostyaSha** [comment]
*[src/main/java/com/github/dockerjava/core/command/BuildImageCmdImpl.java, line 318 \[r1\]](https://reviewable.io:443/reviews/docker-java/docker-java/684#-KQEWHl3ssfwfNxrm7ha-r1-318:-KQGjHZQtLfVKvgLg_zB:bessalq) ([raw file](https://github.com/docker-java/docker-java/blob/761f41655f9e01ed25552309b7dd3a23472a56c4/src/main/java/com/github/dockerjava/core/command/BuildImageCmdImpl.java#L318)):*
<details><summary><i>Previously, cberes (Corey Beres) wrote…</i></summary>
> you  prefer to create the map when the object is created? for example:
> private Map<String, String> labels = new HashMap<>();
</details>
I mean don't use `this` where it not needed. In this case `labels` are not hidden by any input argument, so redundant.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/docker-java/docker-java/684)*

File: src/main/java/com/github/dockerjava/api/command/BuildImageCmd.java [hunk 2]
```diff
@@ -145,6 +151,11 @@
     */
     BuildImageCmd withShmsize(Long shmsize);
 
+    /**
+    *@since {@link RemoteApiVersion#VERSION_1_23}
+    */
+    BuildImageCmd withLabel(String key, String value);
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/api/command/BuildImageCmd.java]: Why only one label? `withLabels(Map<String, String>)`

**Comment by cberes** [comment]
I followed the same pattern as withBuildArg(String, String). So withLabel(String, String) can be called multiple times to add multiple labels. But I will change the method to accept a map if you prefer that. Or have both withLabels(Map) and withLabel(String, String)?

**Comment by KostyaSha** [comment]
I prefer raw data type as docker-java atm is simple layer to API. Your method should be `addLabel` according to logic. But then it become additional logic not related to clean API...

**Comment by cberes** [comment]
ok, I changed `withLabel(String, String)` to `withLabels(Map<String, String>)`

**Comment by KostyaSha** [comment]
rebuilding PR as too much matrix configuration builds failed...

**Comment by KostyaSha** [comment]
Looks fine.

**Pull request merged**